pkg/dmaasbackup/periodic_backup.go

```diff
@@ -78,5 +108,72 @@ func getNextDue(cr cron.Schedule, schedule *v1alpha1.VeleroScheduleDetails, now
 	}
 
 	nextSync := cr.Next(lastSync)
-	return now.After(nextSync), nextSync.Sub(lastSync)
+	return now.After(nextSync), nextSync.Sub(now)
+}
+
+func (d *dmaasBackup) cleanupPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) error {
+	d.logger.Debug("Processing cleanup for schedule")
+
+	if dbkp.Spec.PeriodicFullBackupCfg.FullBackupRetentionThreshold == 0 {
+		d.logger.Debug("Skipping cleanup since FullBackupRetentionThreshold is 0")
+		return nil
+	}
+
+	// delete backups for schedule as per fullBackupRetentionThreshold
+	// we need to retain the backups created by current active schedule and
+	// last 'FullBackupRetentionThreshold' number of schedules.
+	requiredSchedule := dbkp.Spec.PeriodicFullBackupCfg.FullBackupRetentionThreshold + 1
+
+	scheduleCount := len(dbkp.Status.VeleroSchedules)
```

Comment:
this count includes 'Erased' VeleroSchedules also.. those should be probably removed from verifying against retentionThreshold